docs/tutorial.rst

```diff
@@ -673,6 +673,113 @@ breakpoints follows the recombination rate closely.
    :alt: Density of breakpoints along the chromosome.
 
 
+.. _sec_tutorial_dtwf:
+
+*******************************
+Wright-Fisher simulations
+*******************************
+
+Msprime provides the option to perform discrete-time Wright-Fisher simulations
+for scenarios when the coalescent model is not appropriate, including large
+sample sizes, multiple chromosomes, or recent migration. If you use
+``msprime``'s Wright-Fisher simulations in your work, please cite the `paper
+<paper_doi>`, which explains why the coalescent model should not be used in
+these cases.
+
+To use this option, set the flag ``model="dtwf"`` as in the following example::
+
+    >>> tree_sequence = msprime.simulate(
+    ...     sample_size=6, Ne=1000, length=1e4, recombination_rate=2e-8,
+    ...     model="dtwf")
+
+
+All other parameters can be set as usual.
+
+-------------------------------------
+Multiple chromosomes
+-------------------------------------
+
+.. warning:: This option is under active development and is expected to change.
+
+Multiple chromosomes can be simulated by specifying a recombination map with
+hotspots between chromosomes. For example, to simulate two chromosomes each 1
+Morgan in length:
+
+.. code-block:: python
+
+    rho = 1e-8
+    positions = [0, 1e8-1, 1e8, 2e8-1]
+    rates = [rho, 0.5, rho, 0]
+    num_loci = positions[-1]
+
+    recombination_map = msprime.RecombinationMap(
+        positions=positions, rates=rates, num_loci=num_loci)
+
+    tree_sequence = msprime.simulate(
+        sample_size=1000, Ne=10000, recombination_map=recombination_map,
+        model="dtwf")
+
+Care must be taken when simulating whole genomes this way, as the rescaling
+required to model such large fluctuations in recombination rate can result in
+the following error: ``Bad edge interval where right <= left``
+
+This can be avoided by discretizing the genome into 100bp chunks by changing
+the above to:
+
+.. code-block:: python
+
+    rho = 1e-8
+    positions = [0, 1e8-1, 1e8, 2e8-1]
+    rates = [rho, 0.5, rho, 0]
+    num_loci = positions[-1] // 100 # Discretize into 100bp chunks
+
+-------------------------------------
+Hybrid simulations
+-------------------------------------
+
+In some situations Wright-Fisher simulations are desireable yet less
+computationally efficient than coalescent simulations, for example simulating a
+small sample in a recently admixed population. In these cases, a hybrid model
+offers an excellent tradeoff between simulation accuracy and performance.
+
+This involves two steps:
+
+1. Simulating a fixed number of Wright-Fisher generations using the
+   ``__tmp_max_time`` parameter and returning a partial tree sequence (which
```

Comment:
Ugh --- we'll have to fix this to actually work properly and call it ``max_time`` before release.